app/controllers/oneoffs_controller.rb

```diff
@@ -34,6 +37,23 @@ def create_params
     )
   end
 
+  def env_var_params
+    return {} if params[:env_vars].nil?
+
+    raise ExceptionHandler::BadRequest.new("env_vars should be a hash") unless params[:env_vars].is_a? ActionController::Parameters
+
+    params[:env_vars].each do |key, value|
+      if !key.is_a? String
+        raise ExceptionHandler::BadRequest.new("Keys in env_vars should be strings")
+      end
+      if !value.is_a? String
+        raise ExceptionHandler::BadRequest.new("Values in env_vars should be strings")
+      end
+    end
+
+    params[:env_vars].permit(params[:env_vars].keys)
```

Comment:
What does this line do?

Reply:
It turns the hash into a permitted params instance so it will be accepted by activerecord for mass assignment

Reply:
Thanks, I see. 
So it's not permitting, it's a conversion allowing every input from external.

I think it's better to use `permit!` as described in following article to warn that "we allow all the parameters from input here!".

https://www.aloucaslabs.com/miniposts/how-to-permit-all-parameters-in-rails-controller-strong-parameters